python/populate_datatype_fields.py

```diff
@@ -0,0 +1,211 @@
+# /// script
+# requires-python = ">=3.8"
+# dependencies = [
+#     "xnat==0.7.2",
+#     "ismrmrd==1.14.2",
+#     "xmlschema==4.1.0",
+# ]
+# ///
+
+import xnat
+from pathlib import Path
+from datetime import datetime
+from mrd_2_xnat import mrd_2_xnat
+import os
+import ismrmrd
+import logging
+from typing import Any, Tuple
+
+# Configure logging
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
+    handlers=[
+        logging.StreamHandler(),  # Console output
+        logging.FileHandler("xnat_mrd_processing.log"),  # File output
+    ],
+)
+
+logger = logging.getLogger(__name__)
+
+
+def upload_mrd_data(
+    xnat_session,
+    mrd_file_path,
+    project_name,
+    scan_id="cart_cine_scan",
+    experiment_date=None,
```

Comment:
```suggestion
    experiment_date="2022-05-04",
```
You might as well set the default directly here, rather than detecting it as None below to do it